src/OpenLoco/TownManager.cpp

```diff
@@ -103,7 +103,7 @@ namespace OpenLoco::TownManager
             }
 
             // Write new history point.
-            currTown.history[currTown.history_size - 1] = popSteps & 0xFF;
+            currTown.history[std::clamp(currTown.history_size - 1, 0, (int)std::size(currTown.history))] = popSteps & 0xFF;
```

Comment:
Could we split this for readability? e.g.:
```suggestion
            auto histIndex = std::clamp(currTown.history_size - 1, 0, (int)std::size(currTown.history));
            currTown.history[histIndex] = popSteps & 0xFF;
```

Reply:
We should also avoid the C-style cast.

Reply:
done